### DISCUSSION TITLE
Why does relations in nestjs-paginate perform a LEFT JOIN even if the related entities are not filtered/selected?

### DISCUSSION THREAD

**Discussion by estevaodias** [ppetzold/nestjs-paginate#933] [Q&A]
Apparently, if you configure your pagination this way, a `LEFT JOIN` to `tags` is made even when you request `{{baseUrl}}/api/v2/people`. Is it intentional to include a relation without specifying `filter.tags.name` or something similar?

```ts
{
  filterableColumns: {
    name: [FilterOperator.EQ, FilterOperator.IN],
    'tags.name': [FilterOperator.EQ, FilterOperator.IN],
  },
  relations: [
    'tags',
  ],
  select: [
    'tags.id',
  ],
}
```

Here is the output query:

```
SELECT "__root"."id" AS "__root_id", ... LEFT JOIN "people_tags" ... WHERE "__root"."deleted_at" IS NULL
```

**Comment by Helveg** [comment]
If I understand correctly, the `tags` relationship is indeed always included, because you list it in the `relations` array. To allow a user to filter on `tags.name` you don't have to list it there.

AFAIK, if you want to include the relationship only when it's filtered on, then you'll have to inspect the `PaginateQuery` object, to see if it has the `tags.name` filter set, and create your `relations` array accordingly.

The `relations` array is simply the array of relationships to load on the listed entities.
